src/common/libevent.hpp

```diff
@@ -463,6 +466,150 @@ class Bufferevent : public NonCopyable, public NonMovable {
     }
 };
 
+class EvdnsBase {
+  public:
+    // Add a reference to Var<EventBase> base to make sure it is referenced
+    Var<EventBase> evbase = nullptr;
+    evdns_base *dns_base = nullptr;
+
+    EvdnsBase() {}
+    ~EvdnsBase() {}
+
+    template <decltype(evdns_base_new) construct = ::evdns_base_new,
+              decltype(evdns_base_free) destruct = ::evdns_base_free>
+    static Var<EvdnsBase> create(Var<EventBase> base,
+                                 bool initialize_nameservers = true,
+                                 bool fail_requests = true) {
+
+        auto pointer = construct(base->evbase, initialize_nameservers);
+        if (pointer == nullptr) {
+            MK_THROW(EvdnsBaseNewExceptionError);
+        }
+        Var<EvdnsBase> _base(new EvdnsBase, [fail_requests](EvdnsBase *ptr) {
```

Comment:
Again naming convention: I'd not call this variable `_base` (usually leading or trailing underscore is used to refer to object attributes). I understand that you called it `_base` because you already have a `base`. Why not `dns_base`?